pkg/controller/atlasproject/maintenancewindow_test.go

```diff
@@ -1,13 +1,50 @@
 package atlasproject
 
 import (
+	"context"
+	"errors"
 	"testing"
 
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/toptr"
+
+	"github.com/stretchr/testify/require"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/customresource"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+
 	"github.com/stretchr/testify/assert"
 
 	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/project"
 )
 
+type maintenanceWindowClient struct {
+	GetFunc func(projectID string) (*mongodbatlas.MaintenanceWindow, *mongodbatlas.Response, error)
+}
+
+func (c *maintenanceWindowClient) Get(_ context.Context, projectID string) (*mongodbatlas.MaintenanceWindow, *mongodbatlas.Response, error) {
+	return c.GetFunc(projectID)
+}
+
+func (c *maintenanceWindowClient) Update(_ context.Context, _ string, _ *mongodbatlas.MaintenanceWindow) (*mongodbatlas.Response, error) {
+	return nil, nil
```

Comment:
You can use [impl](https://github.com/josharian/impl) which will by default panic unimplemented here. This is what we recommend in (docs/testing.go)[https://github.com/mongodb/mongodb-atlas-kubernetes/blob/main/docs/testing.md]

Not a big deal, but the more similar these mocks are, the less refactor they will need when we move them to their own package.

Reply:
Here I think it's not necessary to apply this once there's no one else consuming these mocks. Next week when I'm back I'll be addressing the ticket to move all the mocks to its own pkg. Also, I will provide "full-featured" mocks together.